pkg/sink/sink_test.go

```diff
@@ -325,13 +330,33 @@ func Test_createResource(t *testing.T) {
 					return nil, err
 				}
 
+				if tt.name != "Namespace" {
```

Comment:
I don't think it is necessary to make make this check. Technically, we are limited in what we create, but conceptually `createResource()` handles anything.

Reply:
Without this check, the Marshal(gotPipelineResource) in the else clause creates additional items like "spec".

Reply:
I see the issue. The labels aren't guaranteed to be ordered. It works if we marshal into an interface (map[string]interface{} underlying type):
```go
var actualResource, expectedResource interface{}
				if err := json.Unmarshal(body, &actualResource); err != nil {
					t.Error(err)
				}
				if err := json.Unmarshal(tt.wantResource, &expectedResource); err != nil {
					t.Error(err)
				}
				if diff := cmp.Diff(expectedResource, actualResource); diff != "" {
					t.Errorf("Diff request body: -want +got: %s", diff)
				}
```

Reply:
OK  I update.